woodwork/schema.py

```diff
@@ -146,6 +153,109 @@ def time_index(self):
+            _validate_not_setting_index_tags(new_semantic_tags, col_name)
+            self.columns[col_name]['semantic_tags'] = new_semantic_tags
+
+            if retain_index_tags and 'index' in original_tags:
+                self._set_index_tags(col_name)
+            if retain_index_tags and 'time_index' in original_tags:
+                self._set_time_index_tags(col_name)
+
+    def add_semantic_tags(self, semantic_tags):
+        """Adds specified semantic tags to columns, updating the Woodwork typing infromation.
+        Will retain any previously set values.
+
+        Args:
+            semantic_tags (dict[str -> str/list/set]): A dictionary mapping the columns
+                in the DataFrame to the tags that should be added to the column's semantic tags
+        """
+        _check_semantic_tags(self.columns.keys(), semantic_tags)
+        for col_name, tags_to_add in semantic_tags.items():
+            tags_to_add = _convert_input_to_set(tags_to_add)
+            _validate_not_setting_index_tags(tags_to_add, col_name)
+            new_semantic_tags = _add_semantic_tags(tags_to_add, self.semantic_tags[col_name], col_name)
+            self.columns[col_name]['semantic_tags'] = new_semantic_tags
+
+    def remove_semantic_tags(self, semantic_tags):
+        """Remove the semantic tags for any column names in the provided semantic_tags
+        dictionary, updating the Woodwork typing information. Including `index` or `time_index`
+        tags will set the Woodwork index or time index to None for the DataFrame.
+
+        Args:
+            semantic_tags (dict[str -> str/list/set]): A dictionary mapping the columns
+                in the DataFrame to the tags that should be removed to the column's semantic tags
+        """
+        _check_semantic_tags(self.columns.keys(), semantic_tags)
+        for col_name, tags_to_remove in semantic_tags.items():
+            standard_tags = self.logical_types[col_name].standard_tags
+            tags_to_remove = _convert_input_to_set(tags_to_remove)
+
+            new_semantic_tags = _remove_semantic_tags(tags_to_remove,
+                                                      self.semantic_tags[col_name],
+                                                      col_name,
+                                                      standard_tags,
+                                                      self.use_standard_tags)
+            # If the index is removed, reinsert any standard tags not explicetly removed
+            original_tags = self.semantic_tags[col_name]
+            if self.use_standard_tags and 'index' in original_tags and 'index' not in new_semantic_tags:
+                standard_tags_removed = tags_to_remove.intersection(standard_tags)
+                standard_tags_to_reinsert = standard_tags.difference(standard_tags_removed)
+                new_semantic_tags = new_semantic_tags.union(standard_tags_to_reinsert)
```

Comment:
This is some new logic that doesn't exist on DataTables: If you remove the `index` tag, previously we weren't getting back the standard tags, but that seems wrong to me that an `Integer` column that was previously an index wouldn't get its `'numeric'` tag back once the index was removed.

Reply:
Good catch. I agree that this fix should be implemented!